### DISCUSSION TITLE
Is there a "Pillow compatible" API for dealing with animated images?

### DISCUSSION THREAD

**Discussion by maj113** [bigcat88/pillow_heif#295] [Q&A] [answered]
in my projects I employ `image.is_animated` and later on something like:

```py
        for i in range(self.frame_len):
            self.img.seek(i)
            raw_frames.append(self.img.copy())

        frame_durations = [duration if (duration := frame.info.get('duration', 100)) else 100 for frame in raw_frames] 
 ```

This works fine for almost all animated image formats both built in pillow and 3rd party (APNG/WEBP/AVIF/...) however not for heif

image used: https://nokiatech.github.io/heif/content/image_sequences/starfield_animation.heic

**Comment by bigcat88** [comment, marked as answer]
AFAIK, `libheif` does not support such images(it supports only first image in sequence):

https://github.com/strukturag/libheif/issues/57
https://github.com/strukturag/libheif/issues/475

**Reply by maj113** [reply]
Ah I see, that's a shame, thanks for clarifying
